File: framework/play/src/main/java/play/data/validation/ValidationError.java
package play.data.validation;

import java.util.*;

/**
 * A validation error.
 */
public class ValidationError {
    
    private String key;
    private String message;
    private List<Object> arguments;
    
    /**
     * Construct a new ValidationError.
     *
     * @param key The error key.
     * @param message The error message.
     * @param arguments The error message arguments.
     */
    public ValidationError(String key, String message, List<Object> arguments) {
        this.key = key;
        this.message = message;
        this.arguments = arguments;
    }
    
    /**
     * Retrieve the error key.
     */
    public String key() {
        return key;
    }
    
    /**
     * Retrieve the error message.
     */
    public String message() {
        return message;
    }
    
    /**
     * Retrieve the error arguments.
     */
    public List<Object> arguments() {
        return arguments;
    }
    
    public String toString() {
        return "ValidationError(" + key + "," + message + "," + arguments + ")";
    }
    
}

File: framework/src/play-datacommons/src/main/scala/play/api/data/validation/ValidationError.scala
/*
 * Copyright (C) 2009-2016 Lightbend Inc. <https://www.lightbend.com>
 */
package play.api.data.validation

/**
 * A validation error.
 *
 * @param messages the error message, if more then one message is passed it will use the last one
 * @param args the error message arguments
 */
case class ValidationError(messages: Seq[String], args: Any*) {

  lazy val message = messages.last

}

object ValidationError {

  def apply(message: String, args: Any*) = new ValidationError(Seq(message), args: _*)

}
